### ISSUE TITLE
EclipseLink NoSQL support

### ISSUE THREAD

**Issue by ratcashdev** [payara/Payara#368]
Adding the http://central.maven.org/maven2/org/eclipse/persistence/org.eclipse.persistence.nosql/2.6.0/org.eclipse.persistence.nosql-2.6.0.jar
jar to Payara's (4.1.152.1) glassfish/modules directory seems to be OK, but afterwards I am getting the exception: org.eclipse.persistence.nosql.adapters.mongo.MongoPlatform cannot be cast to org.eclipse.persistence.internal.databaseaccess.DatabasePlatform

adding the older (2.5.2) jar to plain glassfish works. I suspect this may be due to the signed/unsigned nature of the jars. Where can I download the jar built from the patched eclipselink source used for Payara?

**Comment by smillidge** [comment]
Patched jars are here https://github.com/payara/Payara_PatchedProjects/tree/master/org/eclipse/persistence

**Comment by ratcashdev** [comment]
I don't see the nosql jar there. Would it be possible to upload? I'd like to rule out the hypothesis above.

**Comment by davewinters** [comment]
Hi,

At present the eclipselink nosql jar is  not usually bundled as part of the EclipseLink with Glassfish 4.1  or Payara. I will upload this jar file to out patched_projects repository and let you know once it has been just so you can test. Could you send us the server log file which contains the exception: org.eclipse.persistence.nosql.adapters.mongo.MongoPlatform cannot be cast to org.eclipse.persistence.internal.databaseaccess.DatabasePlatform

Thanks.

**Comment by ratcashdev** [comment]
This is the stack trace.
```
Info:   EclipseLink, version: Eclipse Persistence Services - 2.6.0.payara-p1
Info:   /file:/.../myproject/WEB-INF/classes/_mongo login successful
Info:   /file:/.../myproject/WEB-INF/classes/_mongo logout successful
Severe:   java.lang.ClassCastException: org.eclipse.persistence.nosql.adapters.mongo.MongoPlatform cannot be cast to org.eclipse.persistence.platform.database.DatabasePlatform
	at org.eclipse.persistence.internal.sessions.DatabaseSessionImpl.getPlatform(DatabaseSessionImpl.java:491)
	at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.deploy(EntityManagerSetupImpl.java:730)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.getAbstractSession(EntityManagerFactoryDelegate.java:205)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:305)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
	at org.glassfish.persistence.jpa.JPADeployer$2.visitPUD(JPADeployer.java:451)
	at org.glassfish.persistence.jpa.JPADeployer$PersistenceUnitDescriptorIterator.iteratePUDs(JPADeployer.java:510)
	at org.glassfish.persistence.jpa.JPADeployer.iterateInitializedPUsAtApplicationPrepare(JPADeployer.java:492)
	at org.glassfish.persistence.jpa.JPADeployer.event(JPADeployer.java:398)
	at org.glassfish.kernel.event.EventsImpl.send(EventsImpl.java:131)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:487)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:219)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:491)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:539)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:535)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:534)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:557)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:556)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1464)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:109)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1846)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1722)
	at com.sun.enterprise.v3.admin.AdminAdapter.doCommand(AdminAdapter.java:534)
	at com.sun.enterprise.v3.admin.AdminAdapter.onMissingResource(AdminAdapter.java:224)
	at org.glassfish.grizzly.http.server.StaticHttpHandlerBase.service(StaticHttpHandlerBase.java:189)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:206)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:180)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:283)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:200)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:132)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:111)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:536)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:745)
```

The persistence unit is defined as:
```
<persistence-unit name="mongo" transaction-type="JTA">
    <class>com.mypackage.MongoClass1</class>
    <class>com.mypackage.MongoClass2</class>
	<exclude-unlisted-classes>true</exclude-unlisted-classes>
    <properties>
      <property name="eclipselink.target-database" value="org.eclipse.persistence.nosql.adapters.mongo.MongoPlatform"/>
      <property name="eclipselink.nosql.connection-spec" value="org.eclipse.persistence.nosql.adapters.mongo.MongoConnectionSpec"/>
      <property name="eclipselink.nosql.property.mongo.port" value="27017"/>
      <property name="eclipselink.nosql.property.mongo.host" value="192.168.1.24"/>
      <property name="eclipselink.nosql.property.mongo.db" value="collection"/>
	  <property name="eclipselink.nosql.property.user" value="user"/>
	  <property name="eclipselink.nosql.property.password" value="pw"/>
      <!--<property name="eclipselink.logging.level" value="FINEST"/>-->
    </properties>
  </persistence-unit>
```

**Comment by ratcashdev** [comment]
The full server log is in: https://gist.github.com/ratcashdev/b00107d77fdbc5c53b00

**Comment by davewinters** [comment]
Ok thanks. I will upload the nosql eclipselink jar and update you once done.

**Comment by ratcashdev** [comment]
Please don't forget about this.

**Comment by franciscocpg** [comment]
@ratcashdev 
In fact this problem refers to this bug https://bugs.eclipse.org/bugs/show_bug.cgi?id=462125

**Comment by franciscocpg** [comment]
I was facing the same problem with payara-micro-4.1.153 due to the bug mentioned in the previous comment and I solved it by creating a [patch](https://github.com/franciscocpg/payara-patchs/raw/master/payara-micro-4.1.153-eclipse-link-2.6.1-RC1.jar) with eclipse link 2.6.1.

**Comment by andreasernst** [comment]
I got this issue with MariaDB. This happens if i redeploy an application with EclipseLink 2.6.1. Reload of the application does not help. You need to restart Payara.

**Comment by mikecroft** [comment]
Is this still an issue with the latest Payara Server? 162? The version of Eclipselink currently provided is a patched version of 2.6.2.

@franciscocpg if your patch is still relevant and necessary, would you be prepared to [sign our CLA](https://github.com/payara/Payara/blob/master/PayaraCLA.pdf) so that we can incorporate your change?

**Comment by mikecroft** [comment]
The latest version of Payara packages a patched EclipseLink 2.6.3 which should resolve this according to previous comments. Please reopen with details if not.
